pkg/operator/controllers/routefix/routefix_controller.go

```diff
@@ -0,0 +1,60 @@
+package routefix
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	securityclient "github.com/openshift/client-go/security/clientset/versioned"
+	"github.com/sirupsen/logrus"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/Azure/ARO-RP/pkg/operator/controllers"
+)
+
+//RouteFixReconciler is the controller struct
+type RouteFixReconciler struct {
+	kubernetescli kubernetes.Interface
+	configcli     configclient.Interface
+	seccli        securityclient.Interface
+	log           *logrus.Entry
+}
+
+//NewReconciler creates a new Reconciler
+func NewReconciler(log *logrus.Entry, kubernetescli kubernetes.Interface, configcli configclient.Interface, seccli securityclient.Interface) *RouteFixReconciler {
+
+	return &RouteFixReconciler{
+		log:           log,
+		kubernetescli: kubernetescli,
+		configcli:     configcli,
+		seccli:        seccli,
+	}
+}
+
+//Reconcile fixes the daemonset Routefix
+func (r *RouteFixReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	_, err := r.kubernetescli.AppsV1().DaemonSets(kubeNamespace).Get(ctx, "routefix", metav1.GetOptions{})
+	if err != nil && !errors.IsAlreadyExists(err) {
+		rf := NewRouteFix(r.log, r.kubernetescli, r.seccli)
```

Comment:
Can we just make it part of the RouteFixReconciler? Or drop this side struct at all and do everything in `RouteFixReconciler`